target/scripts/helper-functions.sh

```diff
@@ -225,33 +225,33 @@ export -f _notify
 CHKSUM_FILE=/tmp/docker-mailserver-config-chksum
 
 # Compute checksums of monitored files,
-# returned output is lines of hashed content + filepath pairs.
+# returned output on `stdout`: hash + filepath tuple on each line
 function _monitored_files_checksums
 {
   # If a wildcard path pattern (or an empty ENV) would yield an invalid path
   # or no results, `shopt -s nullglob` prevents it from being added.
   shopt -s nullglob
+  declare -a CERT_FILES
 
   # React to any cert changes within the following letsencrypt locations:
-  local CERT_FILES=(
+  CERT_FILES=(
     /etc/letsencrypt/live/"${SSL_DOMAIN}"/*.pem
     /etc/letsencrypt/live/"${HOSTNAME}"/*.pem
     /etc/letsencrypt/live/"${DOMAINNAME}"/*.pem
   )
 
-  # CERT_FILES should expand to separate paths, not a single string;
-  # otherwise fails to generate checksums for these file paths.
-  #shellcheck disable=SC2068
-  (
-    cd /tmp/docker-mailserver || exit 1
-    exec sha512sum 2>/dev/null -- \
```

Comment:
I take it we don't care about the removed directory check and directory change because `sha512sum` will handle invalid paths fine by ignoring them, and we're returning the output instead of redirecting to a file right? (which original may have done, I haven't gone through history to confirm).

The subshell was only added because of pairing it with the directory change failure case?

Reply:
There is still a directory check. See line 243-246.

>and we're returning the output instead of redirecting to a file right?

The logic haven't changed. See line 51.